### PR TITLE
Vue: Fix memory allocation for useAuthenticator composable

### PR DESCRIPTION
**Pull request by ErikCH**
*Issue #, if available:*
#1050 

*Description of changes:*
Added wrapper to handle memory better when reusing the `useAuthenticator` composable in multiple vue instances. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/vue/src/composables/useAuth.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { ref, reactive, Ref, watchEffect } from 'vue';
 import { getServiceFacade } from '@aws-amplify/ui';
 import { facade } from './useUtils';
 import { InterpretService } from '@/components';
+import { createSharedComposable } from '@vueuse/core';
```
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:6]: That's an awesome library 👀

**Review by wlee221 — APPROVED** [review]

File: packages/vue/src/composables/useAuth.ts (hunk 1 above)
Comment (ErikCH) [reply packages/vue/src/composables/useAuth.ts:6]: It is! Lot's of great utilities in it!

**Comment by reesscot** [comment]
Mind adding a little more detail in the description as to how this improves memory useage? Link to blog article or something?

**Comment by ErikCH** [comment]
@reesscot I showed a pattern of using external composables on my YouTube channel and a commenter mentioned that I should look into how memory is being affected, since he's seen memory leaks with composables that use reactive properties.

After further research I learned about [effectScope](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0041-reactivity-effect-scope.md) and how it helps solve this problem by disposing `computed` and `watch` elements outside of a Vue instance.

I then looked at how the popular [VueUse](https://github.com/vueuse/vueuse) library was creating it's composables. It was using `useEffect` in several places. I then saw it had it's own `createSharedComposable` utility that wrapped itself around `effectScope`.

After adding this into my code base, it seemed that the `watchEffect` in `useAuthenticator` was being called a lot less than it did before.

**Pull request merged**